bitbucket-pipelines.yml

```diff
@@ -0,0 +1,111 @@
+# These are yaml anchors, reused later
+x-bash-env-steps:
+  - &bash_env_export export BASH_ENV="$BITBUCKET_CLONE_DIR/.bashrc"
+  - &bash_env_source source $BASH_ENV
+
+# This is a larger yaml anchor, reused for each pipeline
+default_steps: &default_steps
+  - step:
+      name: Configure Environment Variables
+      services:
+          - docker
+      caches:
+          - docker
+      script:
+        - *bash_env_export
+        - export CI_BRANCH=${BITBUCKET_BRANCH:-master}
+        - export CI_BUILD_NUMBER=$BITBUCKET_BUILD_NUMBER
+        - export CI_PROJECT_USERNAME=$BITBUCKET_REPO_OWNER
+        - export CI_PROJECT_REPONAME=$BITBUCKET_REPO_SLUG
+        - export CI_PROJECT_NAME=$BITBUCKET_REPO_FULL_NAME
+        - /build-tools-ci/scripts/set-environment
+      artifacts:
+        - .bashrc
+  - parallel:
+    - step:
+        name: Static Tests
+        services:
+          - docker
+        caches:
+          - docker
+          - composer
+        script:
+          - *bash_env_export
+          - *bash_env_source
+          - ./.ci/test/static/run
+    - step:
+        name: Build PHP
+        services:
+          - docker
+        caches:
+          - docker
+          - composer
+        script:
+          - *bash_env_export
+          - *bash_env_source
+          - ./.ci/build/php
+        artifacts:
+          - web/**
+          - vendor/**
+  - step:
+      name: Deploy to Pantheon
+      services:
+          - docker
+      caches:
+          - docker
+      script:
+        - *bash_env_export
+        - *bash_env_source
+        - ./.ci/deploy/pantheon/dev-multidev
+      artifacts:
+        - web/**
+        - vendor/**
+  - step:
+      name: Test Visual Regression
+      image: backstopjs/backstopjs:4.1.9
+      services:
+          - docker
+      caches:
+          - docker
+          - node
+      variables:
+        CI_BUILD_URL: $CI_PIPELINE_IID
+        CI_REPOSITORY_URL: $CI_MERGE_REQUEST_IID
+        ARTIFACTS_DIR_URL: https://gitlab.com/api/v4/projects/$CI_PROJECT_ID/jobs/$CI_JOB_ID/artifacts
```

Comment:
replace reference to gitlab here

Reply:
The artifacts logic in the CI script is hard-coded for CircleCI. I think the best approach will be to move the artifact steps to each CI providers config and avoid having CircleCI specific steps in the .ci directory scripts, making those as agnostic as possible